bare-metal/apple-silicon/troubleshooting/cant-connect-using-vnc.mdx

```diff
@@ -20,17 +20,60 @@ categories:
+VNC (Virtual Network Computing) is a system that lets users remotely access and control another computer's desktop. For those using Scaleway's rented Mac mini M1, VNC can be activated by turning on either the "Screen Sharing" or "Remote Management" options in the system preferences. Once set up, you can connect to the Mac mini M1 from another device using its IP address or hostname through VNC clients like RealVNC, TightVNC, and others.
+
+However, some users who've updated their macOS have faced issues with this feature, receiving error messages like `Connection failed to <M1s_address>` and `Make sure Screen Sharing or Remote Management is enabled on the remote computer` under the Sharing section of System Preferences. 
+
+Upon further analysis, we found a design flaw in macOS 13 and 14, where brute-force attacks performed by malicious actors could result in the service halting. [Find out more about this issue on our blog](https://www.scaleway.com/en/blog/m1-how-a-design-flaw-slowed-down-apples-vnc/).
+
+As a precaution, every Mac mini comes with a preinstalled `fail2ban` service. For added security and to ensure VNC service reliability, you can restrict access to macOS Screen Sharing to certain static IPs. This can be done by tweaking your network settings and using macOS's inbuilt firewall, `pf` (Packet Filter), to manage incoming connections.
 
 <Message type="note">
   This is a known issue that multiple Apple users have reported over time. Unfortunately, Apple's support team has not effectively addressed it yet.
 </Message>
 
-If you are experiencing this issue, you can try the following solutions:
+## Enable and Configure pf (Packet Filter)
+
+### Turn on the firewall
+
+macOS comes with a built-in firewall called `pf`. If you haven't already, enable the macOS firewall by going to **System Settings** -> **Network** -> **Firewall**. Toggle the switch to **On** to enable it.
+
+### Configure the firewall
+
+You have to create a custom rule set for `pf` to only allow incoming Screen Sharing connections from your static IP.
 
-## Connect to the Mac mini M1 via SSH and restart Screen Sharing
+1. Open a **Terminal**.
+2. Backup the original `pf.conf` file by copying it to a different location.
+   ```
+   sudo cp /etc/pf.conf /etc/pf.conf.backup
+   ```
+3. Edit the `pf.conf` file with a command like `sudo nano /etc/pf.conf`.
+4. Add the following lines to write rules that block all incoming connections on the Screen Sharing port, except for the desired IP.
+    ```
+    scrub-anchor "com.apple/*"
+    nat-anchor "com.apple/*"
+    rdr-anchor "com.apple/*"
+    dummynet-anchor "com.apple/*"
+    anchor "com.apple/*"
+    load anchor "com.apple" from "/etc/pf.anchors/com.apple"
+    block drop in proto { tcp, udp } from <fail2ban> to any
```

Comment:
Any fail2ban-related rule may be dynamically loaded at machine startup, did you test this including a machine reboot?